src/hotspot/os/posix/vmError_posix.cpp

```diff
@@ -101,15 +101,8 @@ address VMError::get_resetted_sighandler(int sig) {
 }
 
 static void crash_handler(int sig, siginfo_t* info, void* ucVoid) {
-  // unmask current signal
-  sigset_t newset;
-  sigemptyset(&newset);
-  sigaddset(&newset, sig);
-  // also unmask other synchronous signals
-  for (int i = 0; i < NUM_SIGNALS; i++) {
-    sigaddset(&newset, SIGNALS[i]);
-  }
-  PosixSignals::unblock_thread_signal_mask(&newset);
+
+  PosixSignals::unblock_error_signals();
```

Comment:
Just to be clear this crash_handler is only installed for the error signals, so then original code was redundantly processing "sig" twice?

Reply:
I'm unsure what you mean. Do you mean, it had been added twice to newset?

>  sigaddset(&newset, sig);  <<
>  // also unmask other synchronous signals
>  for (int i = 0; i < NUM_SIGNALS; i++) {
>    sigaddset(&newset, SIGNALS[i]);  <<
>  }
>  PosixSignals::unblock_thread_signal_mask(&newset);

That is true. I believe the original coding was from us too, so I feel responsible :)